src/main/java/budgetbuddy/commons/util/CollectionUtil.java

```diff
@@ -50,4 +53,44 @@ public static boolean isAnyNonNull(Object... items) {
         return false;
     }
     //@@author
+
+    /**
+     * Returns a list of all combinations possible for a given list of items.
+     * @return A {@code List} of {@code List}s of the given items.
+     */
+    public static <T> List<List<T>> generateCombinations(List<T> items) {
+        List<List<T>> results = new ArrayList<List<T>>();
+        for (int i = 0; i < items.size(); i++) {
+            results.addAll(generateCombinations(items, i));
+        }
+        return results;
+    }
+
+    /**
+     * Returns a list of r-element combinations possible for a given list of items.
+     * @param r The number of elements for each combination.
+     * @return A {@code List} of size {@code r} {@code List}s of the given items.
```

Comment:
Not sure what you mean by "a list of size r lists of the given items"

Reply:
Probably "a list of size-`r` combinations of items from `items`".

Reply:
Yea, that's correct. I'll tweak it in the future.